sql: TestDistBackfill failed

**Describe the problem**

Seeing TestDistBackfill fail with the following. Happened on both [master](https://teamcity.cockroachdb.com/viewLog.html?buildId=4792543&buildTypeId=Cockroach_UnitTests) and [release-22.1](https://teamcity.cockroachdb.com/viewLog.html?buildId=4792321&buildTypeId=Cockroach_UnitTests).
```
------- Stdout: -------
=== RUN   TestDistBackfill
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistBackfill154976548
    test_log_scope.go:80: use -show-logs to present logs inline
    distsql_plan_backfill_test.go:92: pq: failed to ingest index entries during backfill: addsstable [/Table/107/2/"seven-eight"/78/0,/Table/107/2/"six-zero"/60/0/NULL): error while searching for non-empty span start: cannot read undeclared span /Table/107/2/"six-zero"/60/0/NULL
        declared:
        read local: /Local/Range/Table/107/2/RangeDescriptor at 0,0
        read local: /Local/Lock/Intent/Local/Range/Table/107/2/RangeDescriptor at 0,0
        write global: /Table/107/2/"s{even-eight"/78/0-ix-zero"/60/0/NULL} at 1649128299.561328923,0
        write local: /Local/Lock/Intent/Table/107/2/"s{even-eight"/78/0-ix-zero"/60/0/NULL} at 0,0
        stack:
        goroutine 33146 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x72
        github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset.(*SpanSet).checkAllowed(0xc00474c7e0, 0x0, {{0xc006d49ce0, 0x10, 0x10}, {0x0, 0x0, 0x0}}, 0x6e10980)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset/spanset.go:352 +0x2ed
        github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset.(*SpanSet).CheckAllowed(0xc00474c900, 0x0, {{0xc006d49ce0, 0x10, 0x10}, {0x0, 0x0, 0x0}})
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset/spanset.go:283 +0xd4
        github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset.(*MVCCIterator).checkAllowed(0xc002eeecd0, {{0xc006d49ce0, 0x10, 0x10}, {0x0, 0x0, 0x0}}, 0x1)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset/batch.go:133 +0x1a5
        github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset.(*MVCCIterator).SeekGE(0xc002eeecd0, {{0xc006d49ce0, 0x10, 0x10}, {0x0, 0x0, 0x0}})
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset/batch.go:87 +0x150
        github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.EvalAddSSTable({_, _}, {_, _}, {{0xa86fbf0, 0xc006c40b60}, {{0x16e2e1737465a11b, 0x0, 0x0}, 0x0, ...}, ...}, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable.go:309 +0x1c03
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.evaluateCommand({_, _}, {_, _}, {_, _}, _, {{0x16e2e1737465a11b, 0x0, 0x0}, ...}, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_evaluate.go:493 +0x2c7
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.evaluateBatch({_, _}, {_, _}, {_, _}, {_, _}, _, 0xc00264e800, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_evaluate.go:270 +0xf4f
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evaluateWriteBatchWrapper(_, {_, _}, {_, _}, {_, _}, _, _, {{0x0, ...}, ...}, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_write.go:644 +0x1a7
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evaluateWriteBatchWithServersideRefreshes(_, {_, _}, {_, _}, {_, _}, _, _, {{0x0, ...}, ...}, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_write.go:612 +0x29c
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evaluateWriteBatch(_, {_, _}, {_, _}, _, {{0x0, 0x0, 0x0}, {0x0, ...}}, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_write.go:428 +0x713
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evaluateProposal(0xc002da8800, {0xa76ebc8, 0xc00525c930}, {0xc002702018, 0x8}, 0xc00264e800, {{0x0, 0x0, 0x0}, {0x0, ...}}, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_proposal.go:673 +0x21d
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).requestToProposal(0xc006b28420, {0xa76ebc8, _}, {_, _}, _, {{{0x16e2e17335eb10b2, 0x0, 0x0}, 0x0, ...}, ...}, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_proposal.go:773 +0xfa
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evalAndPropose(0xc002da8800, {0xa76ebc8, _}, _, _, {{{0x16e2e17335eb10b2, 0x0, 0x0}, 0x0, {0x5, ...}, ...}, ...}, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_raft.go:113 +0x205
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeWriteBatch(0xc002da8800, {0xa76ebc8, 0xc00525c930}, 0xc00264e800, 0xc009702d20)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_write.go:167 +0x86b
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeBatchWithConcurrencyRetries(0xc002da8800, {0xa76ebc8, 0xc00525c930}, 0xc00264e800, 0x6e104f0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:477 +0x514
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).sendWithoutRangeID(0xc002da8800, {0xa76ebc8, 0xc00525c900}, 0xc00264e800)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:174 +0x439
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).Send(...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:100
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Send(_, {_, _}, {{{0x16e2e1737465a11b, 0x0, 0x0}, 0x0, {0x5, 0x5, 0x2, ...}, ...}, ...})
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/store_send.go:197 +0x9d6
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).Send(_, {_, _}, {{{0x16e2e1737465a11b, 0x0, 0x0}, 0x0, {0x5, 0x5, 0x2, ...}, ...}, ...})
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/stores.go:191 +0x228
        github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal.func1({0xa76ebc8, 0xc00525c8a0})
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:1006 +0x4d0
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc005064ab0, {0xa76ebc8, 0xc00525c8a0}, {0xc00525c750, 0x650d680}, 0xc008db8d58)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:344 +0x162
        github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal(0xc004fd6400, {0xa76ebc8, 0xc00525c8a0}, {0x0}, 0xc00264e200)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:989 +0x20d
        github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch(0xc004fd6400, {0xa76ebc8, 0xc00525c750}, 0xc00264e200)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:1058 +0x1a5
        github.com/cockroachdb/cockroach/pkg/roachpb._Internal_Batch_Handler.func1({0xa76ebc8, 0xc00525c750}, {0x6b43240, 0xc00264e200})
        	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:9471 +0x89
        github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1({0xa76ebc8, 0xc00525c750}, {0x6b43240, 0xc00264e200}, 0xc0058344a0, 0xc0045be330)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:136 +0x1ca
        google.golang.org/grpc.chainUnaryInterceptors.func1.1({0xa76ebc8, 0xc00525c750}, {0x6b43240, 0xc00264e200})
        	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1116 +0x126
        github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func3({0xa76ebc8, 0xc00525c750}, {0x6b43240, 0xc00264e200}, 0xc0058344a0, 0xc006be8640)
        	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:236 +0xf6
        google.golang.org/grpc.chainUnaryInterceptors.func1.1({0xa76ebc8, 0xc00525c750}, {0x6b43240, 0xc00264e200})
        	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1119 +0x227
        github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({{{0x17}}}, {0xa76ebc8, 0xc00525c750}, {0x6b43240, 0xc00264e200}, 0xc0058344a0, 0xc006be8640)
        	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:71 +0x1c2
        google.golang.org/grpc.chainUnaryInterceptors.func1.1({0xa76ebc8, 0xc00525c750}, {0x6b43240, 0xc00264e200})
        	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1119 +0x227
        github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1.1({0xa76ebc8, 0xc00525c750})
        	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:205 +0x71
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc005064ab0, {0xa76ebc8, 0xc00525c750}, {0x58, 0x203001}, 0xc008db9778)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:344 +0x162
        github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1({0xa76ebc8, 0xc00525c750}, {0x6b43240, 0xc00264e200}, 0xc0058344a0, 0xc006be8640)
        	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:203 +0x136
        google.golang.org/grpc.chainUnaryInterceptors.func1.1({0xa76ebc8, 0xc00525c750}, {0x6b43240, 0xc00264e200})
        	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1119 +0x227
        google.golang.org/grpc.chainUnaryInterceptors.func1({0xa76ebc8, 0xc00525c750}, {0x6b43240, 0xc00264e200}, 0xc0058344a0, 0xc0045be330)
        	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1121 +0x286
        github.com/cockroachdb/cockroach/pkg/roachpb._Internal_Batch_Handler({0x6affee0, 0xc004fd6400}, {0xa76ebc8, 0xc00525c750}, 0xc00474c5a0, 0xc000a11720)
        	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:9473 +0x1db
        google.golang.org/grpc.(*Server).processUnaryRPC(0xc0029a5340, {0xa7e6dc0, 0xc002fa44e0}, 0xc001c9c120, 0xc0028ec660, 0xd226d60, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1282 +0x1515
        google.golang.org/grpc.(*Server).handleStream(0xc0029a5340, {0xa7e6dc0, 0xc002fa44e0}, 0xc001c9c120, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1616 +0xfd4
        google.golang.org/grpc.(*Server).serveStreams.func1.2()
        	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:921 +0xfe
        created by google.golang.org/grpc.(*Server).serveStreams.func1
        	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:919 +0x4d1
    panic.go:661: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistBackfill154976548
--- FAIL: TestDistBackfill (14.89s)
```

--- a/pkg/kv/kvserver/batcheval/cmd_add_sstable.go
+++ b/pkg/kv/kvserver/batcheval/cmd_add_sstable.go
@@ -301,7 +301,7 @@ func EvalAddSSTable(
 	// addition, and instead just use this key-only iterator. If a caller actually
 	// needs to know what data is there, it must issue its own real Scan.
 	if args.ReturnFollowingLikelyNonEmptySpanStart {
-		existingIter := readWriter.NewMVCCIterator(
+		existingIter := spanset.DisableReaderAssertions(readWriter).NewMVCCIterator(
 			storage.MVCCKeyIterKind, // don't care if it is committed or not, just that it isn't empty.
 			storage.IterOptions{UpperBound: reply.RangeSpan.EndKey},
 		)

